scripts/deploy.backend.sh

```diff
@@ -4,28 +4,74 @@ case $ENV in
   "staging")
     ECDSA_KEY_NAME="test_key_1"
     WALLET="cvthj-wyaaa-aaaad-aaaaq-cai"
+    # For security reasons, mainnet root key will be hardcoded in the backend canister.
+    ic_root_key_der="null"
     ;;
   "ic")
     ECDSA_KEY_NAME="key_1"
     WALLET="yit3i-lyaaa-aaaan-qeavq-cai"
+    # For security reasons, mainnet root key will be hardcoded in the backend canister.
+    ic_root_key_der="null"
     ;;
   *)
     ECDSA_KEY_NAME="dfx_test_key"
+    # At the time of writing dfx outputs incorrect JSON with dfx ping (commas between object
+    # entries are missing).
+    # In order to read the root key we grab the array from the '"root_key": [...]' bit, the brackets
+    # to match what candid expects ({}), replace the commas between array entries to match
+    # what candid expects (semicolon) and annotate the numbers with their type (otherwise dfx assumes 'nat'
+    # instead of 'nat8').
+    rootkey_did=$(dfx ping "${ENV:-local}" \
+        | sed -n 's/.*"root_key": \[\(.*\)\].*/{\1}/p' \
+        | sed 's/\([0-9][0-9]*\)/\1:nat8/g' \
+        | sed 's/,/;/g')
+    echo "Parsed rootkey: ${rootkey_did:0:20}..." >&2
+    ic_root_key_der="opt vec $rootkey_did"
     ;;
 esac
 
+II_CANISTER_ID="$(dfx canister id internet_identity --network "$ENV")"
+POUH_ISSUER_CANISTER_ID="$(dfx canister id pouh_issuer --network "$ENV")"
+# URL used by II-issuer in the id_alias-verifiable credentials (hard-coded in II)
+# Represents more an ID than a URL
+II_VC_URL="https://identity.ic0.app"
+# URL used by issuer in the issued verifiable credentials (tipically hard-coded)
+# Represents more an ID than a URL
+# TODO: To be confirmed by DECIDE AI team
+POUH_ISSUER_VC_URL="https://www.elna.ai"
+
 if [ -n "${ENV+1}" ]; then
   dfx deploy backend --argument "(variant {
     Init = record {
          ecdsa_key_name = \"$ECDSA_KEY_NAME\";
          allowed_callers = vec {};
+         supported_credentials = opt vec {
+            record {
+              credential_type = \"ProofOfUniqueness\";
```

Comment:
Seeing it now, it's always "ProofOfUniqueness" - i.e. shouldn't the type be a variant / enum instead of String?

Reply:
Can I create an enum with a set of strings in Candid or Rust?

I would need the mapper of an enum / variant to a string at some point anyway.

That would mean that we'll need to add the types to the enum when we want to support new credentials. Which kind of defeats the purpose to have it as a parameter as well, doesn't it?

Reply:
I think so. In Rust you create an enum which translate to a variant in Candid.

For example this enum: https://github.com/junobuild/juno/blob/e41584c130736ddea8c8e2bd0a444119136599d4/src/libs/shared/src/types.rs#L32C5-L36C6

Becomes this variant in Candid: https://github.com/junobuild/juno/blob/e41584c130736ddea8c8e2bd0a444119136599d4/src/declarations/observatory/observatory.did.d.ts#L10

So I guess here we can have 

```
 #[derive(CandidType, Serialize, Deserialize, Clone)]
pub enum CredentialType {
       ProofOfUniqueness,
}
```

But, I have no clue of the credentials mechanism, just a suggestion if the type are defined in some ways.

Reply:
The credential type in Rust is a String:

https://github.com/dfinity/verifiable-credentials-sdk/blob/a116a2e53a13e4a1f6f7e8429960a7d0d6142e70/rust-packages/ic-verifiable-credentials/src/issuer_api.rs#L94